### ISSUE TITLE
Solid: Warning in development when submitting the form *computations created outside a `createRoot` or `render` will never be disposed*

### ISSUE THREAD

**Issue by MarvinRudolph** [pablo-abc/felte#101]
Hey! 

After I updated to v1.0, I get this warning when submitting the form in development with Solid:
<img width="588" alt="CleanShot 2022-02-19 at 17 46 41@2x" src="https://user-images.githubusercontent.com/6840287/154810291-793444de-48f3-4956-94a4-688242757c94.png">

When I remove the `use:form` I don't get any warning. Even with minimal code and "just" a button, no `onSubmit` function it shows me that warning. Sounds like there is something internally not correct in Felte. 

Any idea why this happens?

Repro:
https://stackblitz.com/edit/vitejs-vite-4pysox

**Comment by pablo-abc** [comment]
Thanks for the reproduction! If I'm seeing correctly, this warning is only getting displayed while submitting. Is this correct?

**Comment by MarvinRudolph** [comment]
> Thanks for the reproduction! If I'm seeing correctly, this warning is only getting displayed while submitting. Is this correct?

Yes, this is correct. This happens only while submitting.

Someone on the Solid Discord mentioned that multiple versions of solid-js could maybe cause that problem. Felte uses `1.2.0` while I have `1.3.x` but I couldn't reproduce it on Stackblitz. Possibly it's still worth to look into that?

**Comment by pablo-abc** [comment]
Solid does not come bundled with Felte! (It's marked as an external dependency, rollup just leaves the import statements) so that should not be an issue.

Chrome's dev tools showed me the warning is being generated here:

https://github.com/pablo-abc/felte/blob/acba6bac0ba40f7e9668ade51b8b97afd7a6481d/packages/core/src/create-form-action.ts#L159

And while yes, that _may_ run outside of a `createRoot` or `render`, I'm handling the disposal manually:

https://github.com/pablo-abc/felte/blob/acba6bac0ba40f7e9668ade51b8b97afd7a6481d/packages/core/src/get.ts#L1-L18

So this seems to be a false positive? Anyway I'd want to prevent this warning from happening so I'll do some more research on this.

**Comment by pablo-abc** [comment]
So this seems to be solved if I don't use Solid's [`observable`](https://www.solidjs.com/docs/latest/api#observable) to transform the signal into an observable and instead I do something like this: 

```typescript
function createSubscriber<T>(signal: () => T) {
  return function subscribe(fn: (data: T) => void) {
    return createRoot((dispose) => {
      createEffect(() => fn(signal()));
      return dispose;
    });
  };
}

const [signal, setSignal] = createSignal();

const observable = {
  subscribe: createSubscriber(signal),
}
```

Essentially implementing my own version of `observable` that uses its own `root`. This was how it was done before (before Solid's `observable` existed) and I had changed it to remove some code. Seems switching back to this is the way to go for now. I'll be pushing a fix during the next few hours. I'll let you know here when its up and we can close this issue then if nothing else pops up!

Thanks a lot for submitting the issue!

**Comment by pablo-abc** [comment]
Just published v1.0.1! The warning is not appearing on my side anymore. Let me know if it's fixed on your side!

**Comment by MarvinRudolph** [comment]
No more warning on my side too. Thanks for fixing!
